src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -2109,6 +2108,9 @@ void JavaThread::start_internal_daemon(JavaThread* current, JavaThread* target,
   target->set_threadOopHandles(thread_oop());
 
   Threads::add(target); // target is now visible for safepoint/handshake
+  // Publish the JavaThread* in java.lang.Thread after the JavaThread* is
+  // on a ThreadsList.
+  java_lang_Thread::set_thread(thread_oop(), target); // isAlive == true now
```

Comment:
Because the `Threads_lock` is grabbed in this function and released again
when the `mu` destructor runs, but I'm going to change my mind.  I've changed
the comment and code to:
```
  // Publish the JavaThread* in java.lang.Thread after the JavaThread* is
  // on a ThreadsList. We don't want to wait for the release when the
  // Theads_lock is dropped when the 'mu' destructor is run since the
  // JavaThread* is already visible to JVM/TI via the ThreadsList.
  java_lang_Thread::release_set_thread(thread_oop(), target); // isAlive == true now
```